app/dot/cpp/repository/repository/BaseRepository.java

```diff
@@ -73,14 +100,27 @@ public List<T> listWithFilterPaginated(Filter filter, int pageSize, int pageNum,
     }
   }
 
-  @NotNull
-  protected static FindOptions getSortOptions(Sort[] sortBy) {
-    return new FindOptions().sort(sortBy);
+  public List<T> listHistory(String trackingId) {
+    final var historyEntities = new ArrayList<T>();
+
+    getHistoryCollection()
+        .find(eq("trackingId", trackingId))
+        .sort(descending("modifiedAt"))
+        .forEach(historyEntities::add);
+
+    return historyEntities;
   }
 
+  // Morphia sets the Mongo codecs based on POJOs, if the Mongo client is used directly then the
+  // codecs have to be set separately, so Morphia is useful in this sense
   @NotNull
-  protected static FindOptions getOptions(int pageSize, int pageNum) {
-    return new FindOptions().skip(pageNum * pageSize).limit(pageSize);
+  private MongoCollection<T> getHistoryCollection() {
+    // todo check for existence?
+    // Mongo will create the collection on the first insert if it does not exist
```

Comment:
Do we really need this comment?

Reply:
this is not really necessary, just good to know and it goes hand in hand with the todo above so check my answer above